cmake/internal.cmake

```diff
@@ -78,10 +78,17 @@ endmacro()
 # This macro cannot be invoked multiple times
 macro(add_leatherman_library)
     include_directories(${${include_var}})
-    add_library(${libname} STATIC ${ARGV})
+    if(LEATHERMAN_SHARED)
+        add_library(${libname} SHARED ${ARGV})
+        target_link_libraries(${libname} ${${deps_var}})
```

Comment:
Might be able to clear `${deps_var}` after doing this. I haven't thought it through.